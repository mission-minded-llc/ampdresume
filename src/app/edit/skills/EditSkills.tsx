"use client";

import { Box, IconButton, Tooltip, Typography } from "@mui/material";

import { EditSkillsSearch } from "./EditSkillsSearch";
import { InfoOutlined } from "@mui/icons-material";
import { LoadingOverlay } from "@/components/LoadingOverlay";
import { MuiLink } from "@/components/MuiLink";
import { ResumeProvider } from "@/components/resume/ResumeContext";
import { SectionTitle } from "../components/SectionTitle";
import { Skills } from "@/components/resume/Skills/Skills";
import { getSkillsForUser } from "@/graphql/getSkillsForUser";
import { useQuery } from "@tanstack/react-query";
import { useSession } from "next-auth/react";

export const EditSkills = () => {
  const { data: session, status } = useSession();

  const isAuthenticatedUser = status === "authenticated" && !!session?.user.id;

  const {
    isPending,
    error,
    data: skillsForUser,
  } = useQuery({
    enabled: isAuthenticatedUser,
    queryKey: ["skillsForUser"],
    queryFn: async () => await getSkillsForUser(session?.user.id),
  });

  if (status === "loading") return <LoadingOverlay message="Loading session..." />;
  if (status === "unauthenticated")
    return (
      <Box>
        Please <MuiLink href="/login">log in.</MuiLink>
      </Box>
    );

  if (isPending) return <LoadingOverlay message="Loading resume data..." />;
  if (error) return <Box>Error loading skills: {error.message}</Box>;

  return (
    <ResumeProvider
      skillsForUser={skillsForUser ?? []}
      companies={[]}
      positions={[]}
      education={[]}
    >
      <SectionTitle title="Add a Skill" />

      <Typography sx={{ mb: 2 }}>Search for a skill to add to your profile:</Typography>
      <EditSkillsSearch />

      <Typography variant="h5" sx={{ mb: 2, mt: 6 }}>
        Your Skills{" "}
        <Tooltip title="Click on a skill to edit or remove it.">
          <IconButton size="small" sx={{ ml: 1 }}>
            <InfoOutlined fontSize="small" />
          </IconButton>
        </Tooltip>
      </Typography>

      <Skills skillType="user" />
    </ResumeProvider>
  );
};
