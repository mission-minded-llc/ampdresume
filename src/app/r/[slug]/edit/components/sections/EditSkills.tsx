import { Box, Container } from "@mui/material";

export const EditSkills = () => {
  return (
    <Container>
      <Box>
        <h1>Edit Skills</h1>
      </Box>
    </Container>
  );
};
