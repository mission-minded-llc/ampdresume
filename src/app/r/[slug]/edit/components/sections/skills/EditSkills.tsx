"use client";

import { Box, Container, Divider, Typography } from "@mui/material";
import { SkillSearch } from "./SkillSearch";
import { Skills } from "@/app/r/[slug]/components/Skills/Skills";

export const EditSkills = () => {
  return (
    <Container>
      <Skills />
      <Divider sx={{ mb: 2, mt: 2 }} />
      <Box>
        <Typography variant="h5" sx={{ mb: 2 }}>
          Add a skill
        </Typography>
        <Typography sx={{ mb: 2 }}>Search for a skill to add to your profile:</Typography>
        <SkillSearch />
      </Box>
    </Container>
  );
};
