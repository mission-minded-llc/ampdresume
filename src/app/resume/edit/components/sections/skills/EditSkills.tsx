"use client";

import { Box, Container, Divider, IconButton, Tooltip, Typography } from "@mui/material";

import { InfoOutlined } from "@mui/icons-material";
import { SkillSearch } from "./SkillSearch";
import { Skills } from "@/components/resume/Skills/Skills";

export const EditSkills = () => {
  return (
    <Container>
      <Divider sx={{ mb: 2, mt: 2 }} />
      <Box>
        <Typography variant="h5" sx={{ mb: 2 }}>
          Add a skill
        </Typography>
        <Typography sx={{ mb: 2 }}>Search for a skill to add to your profile:</Typography>
        <SkillSearch />
      </Box>
      <Divider sx={{ mb: 6, mt: 4 }} />
      <Typography variant="h5" sx={{ mb: 2 }}>
        Your skills{" "}
        <Tooltip title="Click on a skill to edit or remove it.">
          <IconButton size="small" sx={{ ml: 1 }}>
            <InfoOutlined fontSize="small" />
          </IconButton>
        </Tooltip>
      </Typography>
      <Skills skillType="user" />
    </Container>
  );
};
