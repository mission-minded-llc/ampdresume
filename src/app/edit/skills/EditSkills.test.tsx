import "@testing-library/jest-dom";

import { render, waitFor } from "@testing-library/react";

import { EditSkills } from "./EditSkills";
import React from "react";
import { Skill } from "@prisma/client";
import { useQuery } from "@tanstack/react-query";
import { useSession } from "next-auth/react";

jest.mock("next-auth/react", () => ({
  useSession: jest.fn(),
}));

jest.mock("@tanstack/react-query", () => ({
  useQuery: jest.fn(),
}));

jest.mock("@/graphql/getSkillsForUser", () => ({
  getSkillsForUser: jest.fn(),
}));

jest.mock("../components/SectionTitle", () => ({
  SectionTitle: ({ title }: { title: string }) => <div>{title}</div>,
}));

jest.mock("./EditSkillsSearch", () => ({
  EditSkillsSearch: () => <div>EditSkillsSearch</div>,
}));

jest.mock("./SkillsExperience", () => ({
  SkillsExperience: ({ skills }: { skills: Skill[] }) => (
    <div>
      {skills.map((skill) => (
        <div key={skill.id}>{skill.name}</div>
      ))}
    </div>
  ),
}));

jest.mock("@/components/LoadingOverlay", () => ({
  LoadingOverlay: ({ message }: { message: string }) => <div>{message}</div>,
}));

describe("EditSkills", () => {
  const mockSession = {
    user: { id: "user-id" },
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders loading state when session is loading", () => {
    (useSession as jest.Mock).mockReturnValue({ data: null, status: "loading" });
    (useQuery as jest.Mock).mockReturnValue({ isPending: true });

    const { container, getByText } = render(<EditSkills />);

    waitFor(() => expect(getByText("Loading session...")).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });

  it("renders login prompt when unauthenticated", () => {
    (useSession as jest.Mock).mockReturnValue({ data: null, status: "unauthenticated" });
    (useQuery as jest.Mock).mockReturnValue({ isPending: true });

    const { container, getByText } = render(<EditSkills />);

    waitFor(() => expect(getByText("Please log in.")).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });

  it("renders loading state when fetching skills", () => {
    (useSession as jest.Mock).mockReturnValue({ data: mockSession, status: "authenticated" });
    (useQuery as jest.Mock).mockReturnValue({ isPending: true });

    const { container, getByText } = render(<EditSkills />);
    expect(getByText("Loading resume data...")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders error message when fetching skills fails", () => {
    (useSession as jest.Mock).mockReturnValue({ data: mockSession, status: "authenticated" });
    (useQuery as jest.Mock).mockReturnValue({ isPending: false, error: { message: "Error" } });

    const { container, getByText } = render(<EditSkills />);
    expect(getByText("Error loading skills: Error")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders skills when fetched successfully", async () => {
    const mockSkills = [
      { id: "1", name: "Skill 1" },
      { id: "2", name: "Skill 2" },
    ];
    (useSession as jest.Mock).mockReturnValue({ data: mockSession, status: "authenticated" });
    (useQuery as jest.Mock).mockReturnValue({ isPending: false, data: mockSkills });

    const { container, getByText } = render(<EditSkills />);

    await waitFor(() => {
      expect(getByText("Add a Skill")).toBeInTheDocument();
      expect(getByText("Search for a skill to add to your profile:")).toBeInTheDocument();
      expect(getByText("EditSkillsSearch")).toBeInTheDocument();
      expect(getByText("Your Skills")).toBeInTheDocument();
      expect(getByText("Skill 1")).toBeInTheDocument();
      expect(getByText("Skill 2")).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot();
  });

  it("renders no skills message when no skills are found", async () => {
    (useSession as jest.Mock).mockReturnValue({ data: mockSession, status: "authenticated" });
    (useQuery as jest.Mock).mockReturnValue({ isPending: false, data: [] });

    const { container, getByText } = render(<EditSkills />);

    await waitFor(() => {
      expect(getByText("No skills found.")).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot();
  });
});
