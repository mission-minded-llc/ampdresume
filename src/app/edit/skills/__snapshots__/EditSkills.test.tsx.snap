// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditSkills renders error message when fetching skills fails 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    Error loading skills: 
    Error
  </div>
</div>
`;

exports[`EditSkills renders loading state when fetching skills 1`] = `
<div>
  <div>
    Loading resume data...
  </div>
</div>
`;

exports[`EditSkills renders loading state when session is loading 1`] = `
<div>
  <div>
    Loading session...
  </div>
</div>
`;

exports[`EditSkills renders login prompt when unauthenticated 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    Please 
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
      href="/login"
      target="_self"
    >
      log in.
    </a>
  </div>
</div>
`;

exports[`EditSkills renders no skills message when no skills are found 1`] = `
<div>
  <div>
    Add a Skill
  </div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-9ebprp-MuiTypography-root"
  >
    Search for a skill to add to your profile:
  </p>
  <div>
    EditSkillsSearch
  </div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
  >
    No skills found.
  </p>
</div>
`;

exports[`EditSkills renders skills when fetched successfully 1`] = `
<div>
  <div>
    Add a Skill
  </div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-9ebprp-MuiTypography-root"
  >
    Search for a skill to add to your profile:
  </p>
  <div>
    EditSkillsSearch
  </div>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-3i9q64-MuiTypography-root"
  >
    Your Skills
     
    <button
      aria-label="Click on a skill to edit or remove it."
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1mikb9b-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
        data-testid="InfoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
        />
      </svg>
    </button>
  </h5>
  <div>
    <div>
      Skill 1
    </div>
    <div>
      Skill 2
    </div>
  </div>
</div>
`;
